lib/seahorn/HornifyFunction.cc

```diff
@@ -45,53 +47,56 @@ void HornifyFunction::extractFunctionInfo(const BasicBlock &BB) {
   Expr boolSort = sort::boolTy(m_efac);
   ExprVector sorts{boolSort, boolSort, boolSort};
 
-  // memory regions
-  for (const Instruction &inst : BB) {
-    if (const CallInst *ci = dyn_cast<const CallInst>(&inst)) {
-      CallSite CS(const_cast<CallInst *>(ci));
-      const Function *cf = CS.getCalledFunction();
-      if (cf && (cf->getName().equals("shadow.mem.in") ||
-                 cf->getName().equals("shadow.mem.out"))) {
-        const Value &v = *CS.getArgument(1);
-        Expr r = m_sem.symb(v);
-        if (!r)
-          continue;
-        fi.regions.push_back(&v);
-        sorts.push_back(bind::typeOf(r));
+  // memory regions (omitted for partial function stubs)
+  if (!fi.isInferable) {
```

Comment:
splitting everything on `isInferable` makes the code harder to follow. At least add big comment at the beginning. Alternatively, refactor the code. For example, use lambdas to capture big blocks of code, then call it only under appropriate condition.

Reply:
I've made some changes. Is this what you had in mind with respect to lambdas?

Reply:
Yes, but the result is not any more readable than before. Maybe if all the lambda functions are moved up.

Reply:
I've tried rearranging a bit.